src/twisted/web/test/test_cgi.py

```diff
@@ -363,6 +374,44 @@ class CGIScriptTests(unittest.TestCase):
     """
     Tests for L{twcgi.CGIScript}.
     """
+    def tearDown(self):
+        if getattr(self, 'p', None):
+            return self.p.stopListening()
+
+
+    def startServer(self, cgi):
+        root = resource.Resource()
+        cgipath = util.sibpath(__file__, cgi)
+        root.putChild(b"cgi", PythonScript(cgipath))
+        site = server.Site(root)
+        self.p = reactor.listenTCP(0, site)
+        return self.p.getHost().port
+
+
+    def test_urlParameters(self):
+        """
+        If the CGI script is passed URL parameters, dont fall over,
+        as per ticket 9887.
+        """
+        cgiFilename = os.path.abspath(self.mktemp())
+        with open(cgiFilename, 'wt') as cgiFile:
+            cgiFile.write(URL_PARAMETER_CGI)
```

Comment:
This file could also just be a sibling of `test_cgi.py` and then it doesn't need to be written out to a temporary file every time the test runs.

Reply:
Sorry, I'm new to this kind of testing. This construction is used repeatedly in test_cgi.py.